### PR TITLE
feat(gaussdb/cassandra): add flavors data source support

### LINKED ISSUE #1588: [Gaussdb] Add cassandra flavors data source support
I'd like to add flavors data source to cassandra intance.

### PR DESCRIPTION
**Pull request by niuzhenguo**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #1588

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=./huaweicloud/services/acceptance/gaussdb/ TESTARGS='-run=TestAccCassandraFlavorsDataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb/ -v -run=TestAccCassandraFlavorsDataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccCassandraFlavorsDataSource_basic
=== PAUSE TestAccCassandraFlavorsDataSource_basic
=== CONT  TestAccCassandraFlavorsDataSource_basic
--- PASS: TestAccCassandraFlavorsDataSource_basic (38.90s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   38.928s
```

### CODE REVIEW

File: huaweicloud/services/gaussdb/data_source_huaweicloud_gaussdb_cassandra_flavors.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"engine": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "cassandra",
+			},
```
Comment (Lance52259) [inline huaweicloud/services/gaussdb/data_source_huaweicloud_gaussdb_cassandra_flavors.go]: If only `cassandra` supports, maybe we can delete this parameter and write `d.Get("engine").(string)` as `"cassandra"` at ListOpts building.

File: huaweicloud/services/gaussdb/data_source_huaweicloud_gaussdb_cassandra_flavors.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+						},
+						"version": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"az_status": {
+							Type:     schema.TypeMap,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
```
Comment (Lance52259) [inline huaweicloud/services/gaussdb/data_source_huaweicloud_gaussdb_cassandra_flavors.go:65]: maybe we can supports these arguments and use `FilterSliceWithField` to filter the list.

File: docs/data-sources/gaussdb_cassandra_flavors.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+* `memory` - (Optional, String) Specifies the memory size of the flavors.
+
+* `version` - (Optional, String) Specifies the engine version of the flavors.
+
+## Attributes Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - Specifies a data source ID in UUID format.
```
Comment (ShiChangkuo) [inline docs/data-sources/gaussdb_cassandra_flavors.md]: the ID is generated by hashcode package, the value is a hash code not a UUID format.

File: huaweicloud/services/acceptance/gaussdb/data_source_huaweicloud_gaussdb_cassandra_flavors_test.go [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccCassandraFlavorsDataSource_basic(t *testing.T) {
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { acceptance.TestAccPreCheck(t) },
+		Providers: acceptance.TestAccProviders,
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/gaussdb/data_source_huaweicloud_gaussdb_cassandra_flavors_test.go]: suggest to use ProviderFactories

Comment (niuzhenguo) [reply huaweicloud/services/acceptance/gaussdb/data_source_huaweicloud_gaussdb_cassandra_flavors_test.go]: done

File: docs/data-sources/gaussdb_cassandra_flavors.md (hunk 3 above)
Comment (niuzhenguo) [reply docs/data-sources/gaussdb_cassandra_flavors.md]: done

File: huaweicloud/services/gaussdb/data_source_huaweicloud_gaussdb_cassandra_flavors.go (hunk 2 above)
Comment (niuzhenguo) [reply huaweicloud/services/gaussdb/data_source_huaweicloud_gaussdb_cassandra_flavors.go:65]: done

File: huaweicloud/services/gaussdb/data_source_huaweicloud_gaussdb_cassandra_flavors.go (hunk 1 above)
Comment (niuzhenguo) [reply huaweicloud/services/gaussdb/data_source_huaweicloud_gaussdb_cassandra_flavors.go]: done

**Pull request merged**